pkg/controller/core/workload_controller.go

```diff
@@ -166,9 +160,51 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	if ptr.Deref(wl.Spec.Active, true) {
+		var updated bool
+
+		// If a deactivated workload is re-activated we need to reset the RequeueState.
+		if workload.IsEvictedByDeactivation(&wl) && wl.Status.RequeueState != nil {
+			wl.Status.RequeueState = nil
+			updated = true
+		}
+
+		if cond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadRequeued); cond != nil && cond.Status == metav1.ConditionFalse {
+			switch cond.Reason {
+			case kueue.WorkloadEvictedByDeactivation:
+				workload.SetRequeuedCondition(&wl, kueue.WorkloadReactivated, "The workload was reactivated", true)
+				updated = true
+			case kueue.WorkloadEvictedByPodsReadyTimeout:
+				var requeueAfter time.Duration
+				if wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt != nil {
+					requeueAfter = wl.Status.RequeueState.RequeueAt.Time.Sub(r.clock.Now())
+				}
+				if requeueAfter > 0 {
+					return reconcile.Result{RequeueAfter: requeueAfter}, nil
+				}
+				workload.SetRequeuedCondition(&wl, kueue.WorkloadBackoffFinished, "The workload backoff was finished", true)
+				updated = true
+			}
+		}
+
+		if updated {
+			return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
+		}
+	}
+
 	cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
 	if cqOk {
-		if updated, err := r.reconcileSyncAdmissionChecks(ctx, &wl, cqName); updated || err != nil {
+		// because we need to react to API cluster cq events, the list of checks from a cache can lead to race conditions
+		cq := kueue.ClusterQueue{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: cqName}, &cq); err != nil {
+			return ctrl.Result{}, err
+		}
+		// If stopped cluster queue is started we need to set the WorkloadRequeued condition to true.
+		if workload.IsDisabledRequeuedByClusterQueueStopped(&wl) && ptr.Deref(cq.Spec.StopPolicy, kueue.None) == kueue.None {
```

Comment:
The eviction setting on workload_controller here:

```go
if workload.IsAdmitted(wl) {
    if queueStopPolicy != kueue.HoldAndDrain {
        return false, nil
    }
    log.V(3).Info("Workload is evicted because the ClusterQueue is stopped", "clusterQueue", klog.KRef("", cqName))
    workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByClusterQueueStopped, "The ClusterQueue is stopped")
    err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true)
    return true, client.IgnoreNotFound(err)
}
```

After that on jobframework reconciler this event handling and setting Requeued=true.

```go
// 6. handle eviction
if evCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); evCond != nil && evCond.Status == metav1.ConditionTrue {
    if err := r.stopJob(ctx, job, wl, StopReasonWorkloadEvicted, evCond.Message); err != nil {
        return ctrl.Result{}, err
    }
    if workload.HasQuotaReservation(wl) {
        if !job.IsActive() {
            log.V(6).Info("The job is no longer active, clear the workloads admission")
            if evCond.Reason == kueue.WorkloadEvictedByDeactivation ||
                evCond.Reason == kueue.WorkloadEvictedByPodsReadyTimeout ||
                evCond.Reason == kueue.WorkloadEvictedByClusterQueueStopped {
                workload.SetRequeuedCondition(wl, evCond.Reason, evCond.Message, false)
            } else {
                workload.SetRequeuedCondition(wl, evCond.Reason, evCond.Message, true)
            }
            _ = workload.UnsetQuotaReservationWithCondition(wl, "Pending", evCond.Message)
            err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true)
            if err != nil {
                return ctrl.Result{}, fmt.Errorf("clearing admission: %w", err)
            }
        }
    }
    return ctrl.Result{}, nil
}
```

Reply:
Got it, thanks!